arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,48 +65,90 @@ def is_sync(self) -> bool:
         return self.value in self._sync_io_engines
 
 
+time_pattern = re.compile(r"^\d*[d,h,m,s,ms,us]*$")
+
+size_pattern_string = (
+    r"(?:0x\d+)|"
+    r"(?:\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)*)"
+)
+size_pattern = re.compile(
+    rf"^{size_pattern_string}(?:,{size_pattern_string}){{0,2}}$", re.IGNORECASE
+)
+
+size_pattern_with_percent_string = (
+    rf"{size_pattern_string}|(?:[1-9][0-9]?%$|^100%)"
+)
+size_pattern_with_percent = re.compile(
+    rf"^{size_pattern_with_percent_string}$", re.IGNORECASE
+)
+
+size_range_pattern = re.compile(
+    rf"^(?:{size_pattern_string})-(?:{size_pattern_string})$",
+    re.IGNORECASE,
+)
+
+size_multi_range_pattern = re.compile(
+    rf"^(?:{size_pattern_string})-(?:{size_pattern_string})"
+    rf"(?:,(?:{size_pattern_string})-(?:{size_pattern_string})){{0,2}}$",
+    re.IGNORECASE,
+)
+
+
 @dataclass
 class JobParams:
+    # Units
+    kb_base: typing.Annotated[
+        typing.Optional[KbBase],
+        schema.name("Units Base"),
+        schema.description(
+            "Select the interpretation of unit prefixes in input parameters. The "
+            "default is 1024 (compatibility mode): Power-of-2 values with SI prefixes "
+            "(e.g., KB) and Power-of-10 values with IEC prefixes (e.g., KiB). When "
+            "setting this to 1000, inputs comply with IEC 80000-13 and the "
+            "International System of Units (SI): Power-of-2 values with IEC prefixes "
+            "(e.g., KiB) and Power-of-10 values with SI prefixes (e.g., KB)"
+        ),
```

Comment:
I found this text to be confusing.  (Yes, it is fairly clear on a close read, but it would be better if it did not require that level of effort.)

For starters, I would provide the description of the default _after_ the descriptions of the available options.

I think something like this would be better:
> Select the interpretation of unit prefixes in input parameters.  Specifying '1000' indicates that the inputs comply with IEC 80000-13 and the International System of Units (SI), that units with IEC prefixes (e.g., KiB) use power-of-2 multipliers and units with SI prefixes (e.g., KB) use power-of-10 multipliers.  Specifying '1024' indicates the reverse, that units with SI prefixes (e.g., KB) use power-of-2 multipliers and values with IEC prefixes (e.g., KiB) use power-of-10 multipliers.  The default is '1024'.